src/cfg/File.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef _SQUID__SRC_CFG_FILE_H
+#define _SQUID__SRC_CFG_FILE_H
+
+#include "base/Here.h"
+#include "cfg/forward.h"
+#include "sbuf/SBuf.h"
+
+#include <cstdio>
+#include <string>
+
+namespace Cfg
+{
+
+/**
+ * Class used to store required information for the current
+ * configuration file.
+ */
+class File
+{
+public:
+    File(const char *path) : filePath(path) {}
+    ~File();
+
+    /// \return true if the configuration file is open
+    bool isOpen() const { return bool(fd); }
+
+    /// \return the next line to be parsed from this file
+    SBuf nextLine();
+
+    /// \return the configuration file name and line number being processed
+    SourceLocation lineInfo() const { return SourceLocation("parsing", filePath.c_str(), lineNo); }
+
+    /// open and load contents from the file
+    void tryLoadFile();
+
+private:
+    std::string filePath; ///< The file path
```

Comment:
AFAIK this strictly matches the definition of "file path" (two words) used in squid.conf. Wherever squid.conf or included syntax expects a value as a pathname OR filename this string can be written.
  Do you have a suggestion for better name or documentation?

FYI: initial implementation was what you request. But breaks on real-world client installations using pipes. It can even result in files being auto-created under certain OS configurations which break the pipe creation and prevent squid startup.

Reply:
"foo is foo" removal done in be4e9bd

Filtering of the filePath contents moved to the constructor in 4952a5a. I hope that makes your definition of "proper implementation".